src/konserve/core.cljc

```diff
@@ -203,6 +204,13 @@
    (<! (-bassoc store key val))))
 
 
+(defn keys
```

Comment:
I would do the provision of the default value `nil` for the protocol here and just introduce one `-key` protocol then. Since the protocol is not supposed to be used directly from the outside and the behaviour for default arguments should behave the same between implementations this is a cleaner solution.

Reply:
Agreed, done.